src/proteus/dswx_hls.py

```diff
@@ -2250,6 +2507,8 @@ def _compute_hillshade(dem_file, scratch_dir, sun_azimuth_angle,
     """
     shadow_layer_file = tempfile.NamedTemporaryFile(
         dir=scratch_dir, suffix='.tif').name
+    if temp_files_list is not None:
+        temp_files_list.append(shadow_layer_file)
```

Comment:
`temp_files_list` is an optional list that can be provided to some functions so that the calling code knows which temporary files are being created.  The calling code is not required to have a list of temporary files. If `temp_files_list` is `None`, i.e. it's not provided, then nothing needs to be done.

Reply:
Ah, thank you for clarifying.

However, my understanding is that temporary files should be removed before a program terminates. As-written, if a list is not provided, then there is no way for the garbage files to be cleaned up, leading to potentially huge storage space impacts. (For example the worldcover tmp file was often 100MB per granule. If I process 10 granules, that's an extra gig of storage space!)

As something to consider, if the tmp_files_list is not provided, it'd be good to have all temp files deleted automatically during this function. The doc string would then need to be updated to explain this behavior, e.g. "If tmp_files_list is provided, then the path to a tmp file will be appended to the list and not deleted. If tmp_files_list is not provided, then the tmp file will be deleted automatically."

At minimum, let's definitely update the docstring to reflect the current behavior. Maybe something like: 

```
...
       temp_files_list: list (optional)
              Mutable list of temporary files. If provided,
              paths to the temporary files generated will be
              appended to this list.

       Returns
       -------
       hillshade : numpy.ndarray
              Hillshade
              
       Notes
       ------
       All temporary files generated by this function will be
       placed in the `scratch_dir` and they will not be deleted.
       To delete, please supply an argument to `temp_files_list`.
       Paths to the temporary files generated will be appended
       to that list, which the user can use to find and delete the
       temp files.
```

Reply:
Uhm. The file is created using `NamedTemporaryFile`. So, it should be deleted automatically after it's closed by PROTEUS:

"If delete is true (the default), the file is deleted as soon as it is closed. The returned object is always a file-like object whose file attribute is the underlying true file object. This file-like object can be used in a [with](https://docs.python.org/release/3.1.5/reference/compound_stmts.html#with) statement, just like a normal file." 

Source: https://docs.python.org/release/3.1.5/library/tempfile.html#tempfile.NamedTemporaryFile

Thank you for the suggestion of docstrings. I'll update them soon.